scripts/ppr.py

```diff
@@ -0,0 +1,131 @@
+    payload["destination"]["lat"] = stop_places[ii]["lat"]
+    payload["destination"]["lng"] = stop_places[ii]["lng"]
+
+    try:
+        response = requests.post(url, json=payload)
+    except Exception as e:
+        exit(e)
+
+    if response.status_code != 200:
+        exit(f'Request failed with status code {response.status_code}: {response.text}')
+
+    return response.json()
+
+
+def insertPGSQL(cur,json_data,stop_places,path_counter,i,ii):
+    # PPR can return multiple possible paths for one connection:
+    for route in json_data["routes"]:
+        path = route["path"]
+        edges = route["edges"]
+        # distance = route["distance"]
+        relation_id = stop_places[ii]["relation_id"]
+                
+        insertPathsElementsRef(cur, edges, path_counter)
+        insertPath(cur, relation_id, stop_places[i]["IFOPT"], stop_places[ii]["IFOPT"], path)
+
+
+def main():
+    # Connect to PostgreSQL database
+    conn = psycopg2.connect(
+        host = os.environ['host_postgis'],
+        port = "5432",
+        database = "osm2vdv462",
+        user = "admin",
+        password = "admin"
+    )
+
+    # Open a cursor to perform database operations
+    cur = conn.cursor()
+
+    url = 'http://' + os.environ['host_ppr'] + ':8000/api/route'
+    payload = {
+        "start": {
+        },
+        "destination": {
+        },
+        "include_infos": True,
+        "include_full_path": True,
+        "include_steps": True,
+        "include_steps_path": False,
+        "include_edges": True,
+        "include_statistics": True
+    }
+
+    try:
+        with open("profiles/include_accessible.json", "r") as f:
+            payload["profile"] = json.load(f)
+    except Exception as e:
+        conn.close()
+        exit(e)
+
+    stop_places = []
+
+    try:
+        # SRID in POSTGIS is default set to 4326 --> x = lng, Y = lat
+        cur.execute(
+                'SELECT stop_area_osm_id, category, id, ST_X(geom), ST_Y(geom) FROM stop_area_elements'
+            )
+        result = cur.fetchall()
+        for node in result:
+            stop_places.append({"relation_id": node[0], "category": node[1], "IFOPT": node[2], "osm_id": node[2], "osm_type": node[3], "lat": node[4], "lng": node[3]})
+
+    except Exception as e:
+        conn.close()
+        exit(e)
+
+    path_counter = 1
+    
+    print("retrieving paths ...")
+
+    # Iterate through all stop_places to get the path between all of them.
+    for i in range(len(stop_places) - 1):
+        for ii in range(i + 1 , len(stop_places)):
+            
+            json_data = makeRequest(url, payload,stop_places,i,ii)
+            insertPGSQL(cur,json_data,stop_places,path_counter,i,ii)
+            path_counter = path_counter + 1
+            
+            # generate bidirectional paths:
+            json_data = makeRequest(url, payload,stop_places,ii,i)
+            insertPGSQL(cur,json_data,stop_places,path_counter,ii,i)
+            path_counter = path_counter + 1
```

Comment:
Since this will double the amount of requests we will have to make to PPR it would be nice if we could just invert the result we got above. I also fear it might give a different reverse way in some special scenarios, where I'm not sure if this is good or bad. If you have any thoughts on this please let me know.

Inverting should be rather trivial, since we just need to swap the DHIDs and maybe for extra safety we also reverse the geometry using https://postgis.net/docs/ST_Reverse.html (no additional call to `insertPathsElementsRef` needed)

Reply:
> I also fear it might give a different reverse way in some special scenarios, where I'm not sure if this is good or bad.
This was the reason I made the requests in boths directions. If there may be different paths for the both directions, we would not able to depict that. But the only scenarios I can imagine where this could be relevant are staircases with an escalator only in one direction and airports with unidirectional gates/doors.

It is questionable if these special cases are relevant enough to let PPR calculate twice the amount of paths.

Reply:
For now I would stick to your implementation, but adding additional comments in the code describing the special cases this double computation" is catching.